doc/rtd/tutorial/wsl.rst

```diff
@@ -0,0 +1,296 @@
+Server 2022`` environment. If running this tutorial within a virtualized
+environment (`including in the cloud`_), ensure that
+`nested virtualization`_ is enabled.
+
+Install WSL
+===========
+
+.. note::
+    If you have already installed WSL, you can skip this section.
+
+.. code-block:: doscon
+
+    PS> wsl --install
+
+Example output:
+
+.. code-block:: text
+
+    Installing: Virtual Machine Platform
+    Virtual Machine Platform has been installed.
+    Installing: Windows Subsystem for Linux
+    Windows Subsystem for Linux has been installed.
+    Installing: Ubuntu
+    Ubuntu has been installed.
+    The requested operation is successful. Changes will not be effective until the system is rebooted.
+
+Reboot the system when prompted.
+
+Obtain the Ubuntu WSL image
+===========================
+
+Ubuntu 24.04, which is still in development,
+is the first Ubuntu version to support cloud-init in WSL,
+so that is the image that we'll use.
+
+We have two options to obtain the Ubuntu 24.04 WSL image: the Microsoft
+Store and the Ubuntu image server.
+
+Option #1: The Microsoft Store
+------------------------------
+
+If you have access to the Microsoft Store, you can download the
+`Ubuntu 24.04`_ WSL image from within the app.
+
+Click on the "Get" button to download the image.
+
+Once the image has downloaded, do **NOT** click open as that
+will start the instance before we have defined our cloud-init user data
+used to customize the instance.
+
+Once the image has downloaded, you can verify that it is available by
+running the following command:
+
+.. code-block:: doscon
+
+    PS> wsl --list
+
+Example output:
+
+.. code-block:: text
+
+    Windows Subsystem for Linux Distributions:
+    Ubuntu (Default)
+    Ubuntu-Preview
+
+It should show ``Ubuntu-Preview`` in the list of available WSL instances.
+
+Option #2: The Ubuntu image server
+----------------------------------
+
+If the Microsoft Store is not an option, we can instead download the
+Ubuntu 24.04 WSL image from the `Ubuntu image server`_.
+
+Create a directory under the user's home directory to store the
+WSL image and install data.
+
+.. code-block:: doscon
+
+    PS> mkdir ~\wsl-images
+
+Download the Ubuntu 24.04 WSL image.
+
+.. code-block:: doscon
+
+    PS> Invoke-WebRequest -Uri https://cloud-images.ubuntu.com/wsl/noble/current/ubuntu-noble-wsl-amd64-wsl.rootfs.tar.gz -OutFile wsl-images\ubuntu-noble-wsl-amd64-wsl.rootfs.tar.gz
+
+Import the image into WSL storing it in the ``wsl-images`` directory.
+
+.. code-block:: doscon
+
+    PS> wsl --import Ubuntu-Preview wsl-images .\wsl-images\ubuntu-noble-wsl-amd64-wsl.rootfs.tar.gz
+
+Example output:
+
+.. code-block::
+
+    Import in progress, this may take a few minutes.
+    The operation completed successfully.
+
+Create our user data
+====================
+
+User data is the primary way for a user to customize a cloud-init instance.
+Open Notepad and paste the following:
+
+.. code-block:: yaml
+
+    #cloud-config
```

Comment:
We could instead invoke powershell
```
PS> $cloudconfigcontent = @" 
#cloud-config
write_files:
- content: |
    Hello from cloud-init
  path: /var/tmp/hello-world.txt
  permissions: '0777'
"@
PS> New-Item %USERPROFILE%\.cloud-init\Ubuntu-Preview.user-data -ItemType File -Value $cloudconfigcontent

Reply:
Does anybody actually write text files this way (or the `cat << EOF > user-data` method used in bash)? It makes for cool-looking live demos, but I've always thought it's ugly and non-obvious to a newcomer.